Draft-Accepted/feedback_completer_auto_discovery.md

```diff
@@ -0,0 +1,269 @@
+# Auto Discovery for Feedback provider and Tab-Completer
+
+## Motivation
+
+Today, to enable a feedback provider or a tab-completer for a native command,
+a user has to import a module or run a script manually, or do that in their profile.
+There is no way to auto-discover a feedback provider or a tab-completer for a specific native command.
+
+As a tool author, I want to provide a feedback provider or a tab-completer along with my tool installation,
+but I don't want to mess with user's profile to make the feedback provider or tab-completer discoverable.
+
+As a user, I want my feedback providers and tab-completers for the specific native commands to be loaded lazily,
+instead of having to use my profile to load them at session startup.
+
+Relevant GitHub issues:
+- [Enable loading of Feedback providers via config](https://github.com/PowerShell/PowerShell/issues/25115)
+- [Add a way to lazily autoload argument completers](https://github.com/PowerShell/PowerShell/issues/17283)
+
+## Target Scenario
+
+For PowerShell commands (Function or Cmdlet), I presume the completion or feedback support,
+if there is any, will be from the same module.
+So, when the command becomes available, the feedback provider and/or tab-completer will become available too.
+
+Therefore, the auto-discovery for feedback provider and tab-completer targets native commands only.
```

Comment:
Just for the record, there's been a lot of discussion in Discord chat where people are going so far as to use dynamic parameters just to try and create argument completion for standalone scripts. It might be worth considering them as well.